### ISSUE TITLE
[management] transferring ownership of an API to non-existent group makes APIs access unavailbale in UI Console

### ISSUE THREAD

**Issue by apoddany** [gravitee-io/issues#7305]
## :collision: Describe the bug
When transferring the ownership of an API with the rest-api to a non-existent group, APIs won't be accessible anymore in the UI Console with the message "Group[xxx] can not be found."

## :sunrise_over_mountains: To Reproduce
Steps to reproduce the behaviour:
1. In Console UI, create an API
2. From Curl or Postman, try to change the ownership of this API to a non-existing Group : 
```
POST http://localhost:8083/management/organizations/DEFAULT/environments/DEFAULT/apis/34c511f3-2d53-4689-8511-f32d53368917/members/transfer_ownership
{
  "id": "doesnotexist",
  "reference": "null",
  "type": "GROUP",
  "role": "USER"
}
```
3. Check the output of the request : 
```
{
    "message": "Group [doesnotexist] can not be found.",
    "parameters": {
        "group": "doesnotexist"
    },
    "technicalCode": "group.notFound",
    "http_status": 404
}
```
4. Try to access the APIs in the UI Console --> "Group[doesnotexist] can not be found."
5. The fix  is to delete the offending API.  No validation is made prior to the API ownership attribute being updated.

## Current behaviour
A clear and concise description of what is currently happening.

## :movie_camera: Useful information
Screenshot, video, logs, other supporting material

## :computer: Desktop:

***Please complete the following information:***
 - Environment: tested in 3.15.5

## :warning: Potential impacts

***Which other features may be impacted by this fix. This could be populated after fix***

***What are the impacted versions?***

## :link: Dependencies

Link a story or other related things...

**Comment by mouligno** [comment]
@a-cordier isn't linked to #6359 ?

**Comment by mouligno** [comment]
Closed as fixed with #6359
